### PR TITLE
P2PW-1633 - add searchView in SelectTokenFragment

### PR DESCRIPTION
**Pull request by davran1001**
Add support for search tokens in SearchTokenFragment

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt [hunk 1]
```diff
@@ -62,4 +67,32 @@ class SelectTokenFragment : BaseFragment(R.layout.fragment_select_token) {
             emptyTextView.isVisible = isEmpty
         }
     }
+
+    override fun onQueryTextSubmit(query: String?): Boolean = false
+
+    override fun onQueryTextChange(newText: String?): Boolean {
+        val searchText = newText.orEmpty()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt:74]: it looks like that all this filtering and checks should be in presenter, it's his responsibility to define what to show on UI

File: app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt [hunk 2]
```diff
@@ -62,4 +67,32 @@ class SelectTokenFragment : BaseFragment(R.layout.fragment_select_token) {
...
+        if (searchText.length < QUERY_MIN_LENGTH) {
+            tokenAdapter.setItems(tokens)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt:76]: now when we have another type of list of tokens, it's better to name this field as `allTokens`

File: app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt [hunk 3]
```diff
@@ -62,4 +67,32 @@ class SelectTokenFragment : BaseFragment(R.layout.fragment_select_token) {
...
+            return true
+        }
+        val filteredItems = tokens.filter {
+            it.tokenName.startsWith(searchText, ignoreCase = true) || it.tokenSymbol == newText
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt:80]: i think we need to check .contains
tokenName and tokenSymbol .contains newText
to make search more UserFriendly
for now it.tokenSymbol == newText means that only if user will input USDC ... then token will appear

Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt:80]: i think we need to check .contains
tokenName and tokenSymbol .contains newText
to make search more UserFriendly
for now it.tokenSymbol == newText means that only if user will input USDC ... then token will appear

File: app/src/main/res/menu/menu_search.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <item
+        android:id="@+id/menu_search"
+        android:title="@string/search_menu"
+        app:actionLayout = "@layout/item_search_bar"
+        app:showAsAction="always" />
+</menu>
```
Comment (kabramovp2p) [inline app/src/main/res/menu/menu_search.xml:10]: new line

**Pull request merged**